### PR TITLE
Remove redundant location_id validation from agency form

### PR DESCRIPTION
**Pull request by esoergel**
https://manage.dimagi.com/default.asp?258965
@mkangia

### CODE REVIEW

File: corehq/apps/locations/forms.py [hunk 1]
```diff
@@ -334,11 +334,17 @@ def save(self, metadata):
         return location
 
 
+class LocationUserForm(NewMobileWorkerForm):
+    def clean_location_id(self):
+        # This is handled separately
```
Comment (mkangia) [inline corehq/apps/locations/forms.py]: Can we include the reason why we are doing this a little in detail? That would help understand the need of this change quickly and better when someone sees it.

File: custom/enikshay/user_setup.py [hunk 2]
```diff
@@ -266,12 +266,6 @@ def connect_signals():
     commcare_user_post_save.connect(save_user_callback, dispatch_uid="save_user_callback")
 
 
-class ENikshayNewMobileWorkerForm(NewMobileWorkerForm):
-    """Mobile worker list view create modal"""
-    def __init__(self, *args, **kwargs):
-        super(ENikshayNewMobileWorkerForm, self).__init__(*args, **kwargs)
-
-
```
Comment (mkangia) [inline custom/enikshay/user_setup.py]: Why was  `ENikshayNewMobileWorkerForm ` even needed?
You removed it because it was not doing anything special?

**Review by mkangia — APPROVED** [review]
Don't understand the overall impact of the change but otherwise looks good. No blocker, just couple of comments

File: custom/enikshay/user_setup.py (hunk 2 above)
Comment (esoergel) [reply custom/enikshay/user_setup.py]: I think it used to do something which later got moved elsewhere.  At this point it's not doing anything, though.

File: corehq/apps/locations/forms.py (hunk 1 above)
Comment (esoergel) [reply corehq/apps/locations/forms.py]: updated

**Pull request merged**